server/src/main/java/org/opensearch/search/fetch/FetchSearchResult.java

```diff
@@ -82,6 +105,28 @@ public FetchSearchResult fetchResult() {
     public void hits(SearchHits hits) {
         assert assertNoSearchTarget(hits);
         this.hits = hits;
+        if (FeatureFlags.isEnabled(FeatureFlags.PROTOBUF_SETTING) && this.fetchSearchResultProto != null) {
+            QuerySearchResultProto.TotalHits.Builder totalHitsBuilder = QuerySearchResultProto.TotalHits.newBuilder();
+            totalHitsBuilder.setValue(hits.getTotalHits().value);
+            totalHitsBuilder.setRelation(
+                hits.getTotalHits().relation == Relation.EQUAL_TO
+                    ? QuerySearchResultProto.TotalHits.Relation.EQUAL_TO
+                    : QuerySearchResultProto.TotalHits.Relation.GREATER_THAN_OR_EQUAL_TO
+            );
+            FetchSearchResultProto.SearchHits.Builder searchHitsBuilder = FetchSearchResultProto.SearchHits.newBuilder();
+            searchHitsBuilder.setMaxScore(hits.getMaxScore());
+            searchHitsBuilder.setTotalHits(totalHitsBuilder.build());
+            for (SearchHit hit : hits.getHits()) {
+                FetchSearchResultProto.SearchHit.Builder searchHitBuilder = FetchSearchResultProto.SearchHit.newBuilder();
+                searchHitBuilder.setId(hit.getId());
+                searchHitBuilder.setSource(ByteString.copyFrom(hit.getSourceRef().toBytesRef().bytes));
+                searchHitBuilder.setVersion(hit.getVersion());
+                searchHitBuilder.setSeqNo(hit.getSeqNo());
+                searchHitBuilder.setPrimaryTerm(hit.getPrimaryTerm());
+                searchHitsBuilder.addHits(searchHitBuilder.build());
+            }
```

Comment:
You could re-use the method from SearchHits which converts `hits` to `searchHitsProto`

Reply:
Done.